cli/azd/pkg/environment/environment.go

```diff
@@ -202,3 +203,17 @@ func (e *Environment) SetPrincipalId(principalID string) {
 func (e *Environment) GetPrincipalId() string {
 	return e.Values[PrincipalIdEnvVarName]
 }
+
+func normalize(key string) string {
+	return strings.ReplaceAll(strings.ToUpper(key), "-", "_")
```

Comment:
There's some interesting behavior here regarding allowed characters. Currently, any character other than `=` is unrestricted and works correctly with environment loading/saving. However, this would not work correctly in cases when used in a shell.

I suspect we want to restrict naming of project and service to be alphanumeric, underscores and dashes, supporting dash to underscore conversion. Happy to do that separately, just trying to gather feedback on it.

Reply:
> I suspect we want to restrict naming of project and service to be alphanumeric, underscores and dashes, supporting dash to underscore conversion. Happy to do that separately, just trying to gather feedback on it.

I think this makes sense.